Rework the alpha conversion to remove jaggies

arlyon opened the pull request:
The current method doesn't antialias correctly, presumably because the alpha layer is discarded while there is still data. This new method uses wand to handle the background, allowing us also to remove the special case conversion logic.

Also, am I right in saying that  the docstring here is inaccurate? I don't see the kwargs being passed in anywhere. If we were, for example, to support setting the `background` arg, we'd have to support the alpha channel, as transparent is valid. Happy to update the docstring once I get some more context! :)

arlyon commented:
To compare the difference in output, see below (large images, you may need to zoom):

![image](https://user-images.githubusercontent.com/8715352/105607697-3aedb080-5d98-11eb-90f9-6a0173911252.png)

![image](https://user-images.githubusercontent.com/8715352/105607728-68d2f500-5d98-11eb-81b7-c8b401200369.png)

jsvine commented:
Thanks, @arlyon! This seems like a nice improvement. (Note to my future self: The initial setting of "background" came via https://github.com/jsvine/pdfplumber/pull/70.) 

Re. this:

> Also, am I right in saying that the docstring here is inaccurate? I don't see the kwargs being passed in anywhere. If we were, for example, to support setting the `background` arg, we'd have to support the alpha channel, as transparent is valid. Happy to update the docstring once I get some more context! :)

... it appear that you are correct; the docstring is inaccurate. It has been around since 0.5.0, introduced via https://github.com/jsvine/pdfplumber/pull/23; I likely started with a `**kwargs`-based parameter list but scrapped it without updating the docstring. Thank you for catching that.

It sounds like you're more familiar with Wand and image processing than I am. Do you see a particular advantage in users being able to pass `**kwargs` to this method? And where would they be applied? (To `wand.image.Image(...)`?)

And one request: Could you change the target of this PR from `stable` to `develop`?

arlyon committed:
Rework the alpha conversion to remove jaggies

arlyon commented:
There's probably not much of interest except maybe the background colour but if someone want their documents in pink they can open an issue :) haha

I've just deleted the docstring for now, I felt anything I tried to write was fairly self explanatory from the code itself, and I've rebased!

Thanks for the quick reply

jsvine reviewed (approved):
Looks good and makes sense. Thanks, and merging!
